### ISSUE TITLE
fine-tune the grounded-DINO model

### ISSUE THREAD

**Issue by kobic8** [IDEA-Research/Grounded-Segment-Anything#266]
wonderful work! I installed the repo and checked for the case I load images (sea iamges for example) and it detected marine objects such as ships/boats/ jet ski and also the sea-sky and horizon. 
I want to finetune the models to detect the ships and boats by type(their name) given that I have a labaled dataset. I assume that only G-Dino needs to be fine-tuned since the SAM part works fine.
I'd appreciate if you guide me on any other things I'd need to finetune the pretrained models.

**Comment by rentainhe** [comment]
Sorry we did not release the training code for GroundingDINO at this time because our paper is still under review now, we will release it in the future

**Comment by rentainhe** [comment]
> wonderful work! I installed the repo and checked for the case I load images (sea iamges for example) and it detected marine objects such as ships/boats/ jet ski and also the sea-sky and horizon. I want to finetune the models to detect the ships and boats by type(their name) given that I have a labaled dataset. I assume that only G-Dino needs to be fine-tuned since the SAM part works fine. I'd appreciate if you guide me on any other things I'd need to finetune the pretrained models.

For this situation, to recognize specific boats GroundingDINO should be fine-tuned, or you can try to use a close-set detector if you only have limited boats to be recognized

**Comment by kobic8** [comment]
> > wonderful work! I installed the repo and checked for the case I load images (sea iamges for example) and it detected marine objects such as ships/boats/ jet ski and also the sea-sky and horizon. I want to finetune the models to detect the ships and boats by type(their name) given that I have a labaled dataset. I assume that only G-Dino needs to be fine-tuned since the SAM part works fine. I'd appreciate if you guide me on any other things I'd need to finetune the pretrained models.
> 
> For this situation, to recognize specific boats GroundingDINO should be fine-tuned, or you can try to use a close-set detector if you only have limited boats to be recognized

hello @rentainhe  and thank you for the reply. I know I can train a custom a detector. Since I am amazed from the results of the hybrid G-DINO-SAM I'd like just to fine tune it to recognize more specific classes (labels). Do you have any estimation for code release / guidelines of how to fine-tune this model?

**Comment by rentainhe** [comment]
> > > wonderful work! I installed the repo and checked for the case I load images (sea iamges for example) and it detected marine objects such as ships/boats/ jet ski and also the sea-sky and horizon. I want to finetune the models to detect the ships and boats by type(their name) given that I have a labaled dataset. I assume that only G-Dino needs to be fine-tuned since the SAM part works fine. I'd appreciate if you guide me on any other things I'd need to finetune the pretrained models.
> > 
> > 
> > For this situation, to recognize specific boats GroundingDINO should be fine-tuned, or you can try to use a close-set detector if you only have limited boats to be recognized
> 
> hello @rentainhe and thank you for the reply. I know I can train a custom a detector. Since I am amazed from the results of the hybrid G-DINO-SAM I'd like just to fine tune it to recognize more specific classes (labels). Do you have any estimation for code release / guidelines of how to fine-tune this model?

Yes, we may release some fine-tune guide after the paper reviewing, but at this time we did not clean the code yet, it's not ready for release, very sorry

**Comment by kobic8** [comment]
thanks so much! I will wait. if you desire, I can use this "beta version" of your code,  and give some remarks:)
